core/src/main/java/org/mule/runtime/core/processor/chain/ModuleOperationMessageProcessorChainBuilder.java

```diff
@@ -125,16 +131,23 @@ protected Event doProcess(Event event) throws MuleException {
       if (!returnsVoid) {
         return from(publisher)
             .concatMap(request -> just(request)
-                .map(event -> createEventWithParameters(event))
-                .transform(s -> super.apply(s))
+                .map(this::createEventWithParameters)
+                .transform(super::apply)
                 .map(result -> createNewEventFromJustMessage(request, result)));
       } else {
         return publisher;
       }
     }
 
     private Event createNewEventFromJustMessage(Event request, Event response) {
-      return Event.builder(request).message(builder(response.getMessage()).build()).build();
+      final Event.Builder builder = Event.builder(request);
+      if (target.isPresent()) {
+        builder.addVariable(target.get(), response.getMessage().getPayload().getValue(),
```

Comment:
The whole message needs to go as value for the target variable, not just the payload